### DISCUSSION TITLE
Customizable Y Sorting

### DISCUSSION THREAD

**Discussion by wareya** [godotengine/godot-proposals#4541] [2D]
I have a [roguelike](https://github.com/wareya/holyswordseeker) where my entities' `global_position`s and the locations at which they render are slightly different because of animation interpolation (not related to the current physics interpolation stuff, completely custom). So, when using a TileMap with y sorting enabled, entities that pass across objects render on top of or behind them based on their logical location instead of their graphical location, so they pop in and out in situations where they shouldn't, which looks unpolished.

If y-sorting-capable objects checked for a function like `_get_sorting_position()` in their children and called that instead of using its position, **or** if they had a function like `_forward_ysort_position_override(CanvasItem)` (like how TileSet has `_forward_subtile_selection`), I could work around this problem without having to redesign my roguelike's animation system.

**Comment by kinoyaa** [comment]
I also hope to add a customizable y sorting function,I have encountered such a problem recently,I recently designed a block loading system, which needs to instantiate many small scenes in this scene, so it is very troublesome to build

**Comment by kinoyaa** [comment]
[https://github.com/godotengine/godot-proposals/issues/4561](url)I submit my proposal here

**Reply by Zireael07** [reply]
Bad link, 404

**Reply by kinoyaa** [reply]
https://github.com/godotengine/godot-proposals/issues/4561

**Reply by wareya** [reply]
It looks like you're looking for a way to do "deep" y-sorting. I think that's great. I think it should be clear that this is a second type of customization for y-sorting from what I need, though. A new feature that "just" allows deep y-sorting wouldn't make the y-sorting system as customizable as it needs to be.

**Comment by choboy46** [comment]
The current y_sorting is cool and all, but I'm finding it's limited with sorting customizability especially when combined with z_index. Having a custom y_sorting offset would definitely help a lot.
